nextmv/options.py

```diff
@@ -104,24 +151,33 @@ class Options:
         parameter.
     """
 
-    def __init__(self, *parameters: Parameter):
+    def __init__(self, *parameters: Parameter):  # noqa: C901
         """Initializes the options."""
 
         if not parameters:
             return
 
+        self.parameters = copy.deepcopy(parameters)
+
         parser = argparse.ArgumentParser(
             add_help=True,
             usage="%(prog)s [options]",
             description="Options for %(prog)s. Use command-line arguments (highest precedence) "
             + "or environment variables.",
+            allow_abbrev=False,
         )
         params_by_field_name: Dict[str, Parameter] = {}
 
         for p, param in enumerate(parameters):
             if not isinstance(param, Parameter):
                 raise TypeError(f"expected a <Parameter> object, but got {type(param)} in index {p}")
 
+            # See comment below about ipykernel adding a `-f` argument. We
+            # restrict parameters from having the name 'f' or 'fff' for that
+            # reason.
+            if param.name == "f" or param.name == "fff":
+                raise ValueError("parameter names 'f', 'fff' are reserved for internal use")
```

Comment:
This unnecessarily constrains all non-notebook use-cases from using `-f` as a shorthand, right? Potentially not the biggest loss, but it feels odd to me. :sweat_smile:

Reply:
Yes, I agree, it is quite odd. However, this is the workaround that I was able to find.